### PR TITLE
Add wildcard support for header names

### LINKED ISSUE #10811: Add wildcard support for header names in the _cat API
It would be nice to request multiple columns in the _cat API using wildcards like:

    GET _cat/thread_pool?v&h=bulk.*

### PR DESCRIPTION
**Pull request by spinscale**
This adds wildcard support (simple regexes) for specifying header names.
Aliases are supported as well.

Closes #10811

### CODE REVIEW

File: src/main/java/org/elasticsearch/rest/action/support/RestTable.java [hunk 1]
```diff
@@ -96,11 +96,35 @@ public static RestResponse buildTextPlainResponse(Table table, RestChannel chann
...
+                                if (Regex.simpleMatch(header, alias)) {
+                                    headers.add(configuredHeader);
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                } else {
+                    headers.add(header);
+                }
```
Comment (javanna) [inline src/main/java/org/elasticsearch/rest/action/support/RestTable.java]: wondering if extracting this block to a new method would make it a bit more readable, I have no strong opinion though

File: src/main/java/org/elasticsearch/rest/action/support/RestTable.java [hunk 2]
```diff
@@ -96,11 +96,35 @@ public static RestResponse buildTextPlainResponse(Table table, RestChannel chann
...
     }
 
-    private static List<DisplayHeader> buildDisplayHeaders(Table table, RestRequest request) {
+    static List<DisplayHeader> buildDisplayHeaders(Table table, RestRequest request) {
         String pHeaders = request.param("h");
         List<DisplayHeader> display = new ArrayList<>();
         if (pHeaders != null) {
-            for (String possibility : Strings.splitStringByCommaToArray(pHeaders)) {
+            Map<String, Table.Cell> headerMap = table.getHeaderMap();
+            Collection<String> headers = new LinkedHashSet<>(table.getHeaders().size());
```
Comment (javanna) [inline src/main/java/org/elasticsearch/rest/action/support/RestTable.java]: why a set? are duplicates possible?

File: src/test/java/org/elasticsearch/rest/action/support/RestTableTest.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.elasticsearch.rest.action.support.RestTable.buildDisplayHeaders;
+import static org.hamcrest.Matchers.*;
+
+public class RestTableTest extends ElasticsearchTestCase {
+
```
Comment (javanna) [inline src/test/java/org/elasticsearch/rest/action/support/RestTableTest.java]: maybe it is worth to add a couple of REST tests to make sure that everything works on the REST layer, also clients would get tested through it as well.

**Comment by javanna** [comment]
left a few comments, looks good though

File: src/main/java/org/elasticsearch/rest/action/support/RestTable.java (hunk 2 above)
Comment (spinscale) [inline src/main/java/org/elasticsearch/rest/action/support/RestTable.java]: yes, due to selecting 'bulk.*,bulk.*' in the `h` param

File: src/test/java/org/elasticsearch/rest/action/support/RestTableTest.java (hunk 3 above)
Comment (spinscale) [inline src/test/java/org/elasticsearch/rest/action/support/RestTableTest.java]: did it!

File: src/main/java/org/elasticsearch/rest/action/support/RestTable.java (hunk 2 above)
Comment (javanna) [inline src/main/java/org/elasticsearch/rest/action/support/RestTable.java]: right, makes sense

File: src/main/java/org/elasticsearch/rest/action/support/RestTable.java (hunk 1 above)
Comment (spinscale) [inline src/main/java/org/elasticsearch/rest/action/support/RestTable.java]: valid point, also allows to add some javadoc. done

**Comment by javanna** [comment]
LGTM

**Pull request merged**